internal/linker/link_node_modules.bzl

```diff
@@ -127,18 +124,9 @@ def _get_module_mappings(target, ctx):
     for name in _MODULE_MAPPINGS_DEPS_NAMES:
         for dep in getattr(ctx.rule.attr, name, []):
             for k, v in getattr(dep, _ASPECT_RESULT_NAME, {}).items():
-                # A package which was reachable transitively via a *_binary or *_test
-                # rule is assumed to be in the runfiles of that binary,
-                # so we switch the linker target root.
-                # In theory we ought to be able to tell that the files really are
-                # propagated through the runfiles output of the target we are visiting
-                # but it seems that this info is only available in Bazel Java internals.
-                # TODO: revisit whether this is the correct condition after downstream testing
```

Comment:
I like dropping the TODO - just hoping it's not a problem anymore?

Reply:
The distinction isn't important anymore since we don't treat them any differently when passing the modules to the loader as we now rely on runfiles.resolve to resolve to the right place